http/obmc_shell.hpp

```diff
@@ -0,0 +1,214 @@
+#pragma once
+
+#include <pty.h>
+#include <termios.h>
+
+#include <app.hpp>
+#include <boost/asio/posix/stream_descriptor.hpp>
+#include <boost/process/io.hpp>
+#include <websocket.hpp>
+
+#include <map>
+
+namespace crow
+{
+
+namespace obmc_shell
+{
+
+class Handler : public std::enable_shared_from_this<Handler>
+{
+  public:
+    Handler(crow::websocket::Connection* conn) :
+        session(conn), streamFileDescriptor(conn->getIoContext()),
+        doingWrite(false)
+    {}
+
+    ~Handler() = default;
+
+    void doClose()
+    {
+        streamFileDescriptor.close();
+
+        // boost::process::child::terminate uses SIGKILL, need to send SIGTERM
+        int rc = kill(pid, SIGKILL);
+        session = nullptr;
+        if (rc)
+        {
+            return;
+        }
+        waitpid(pid, nullptr, 0);
+    }
+
+    void connect()
+    {
+        pid = forkpty(&ttyFileDescriptor, nullptr, nullptr, nullptr);
+
+        if (pid == -1)
+        {
+            // ERROR
+            if (session)
+            {
+                session->close("Error creating ssh child process");
+            }
+            return;
+        }
+        if (pid == 0)
+        {
+            // CHILD
+
+            // sets the effective user ID
+            // as service user
+            int uidr = setuid(1001);
```

Comment:
This is a step in the right direction.  We should get the UID from the username.  You can use the `getpwnam` function to do this.  For example, as described here: https://stackoverflow.com/questions/39157675/how-to-get-linux-user-id-by-user-name  - but with "service" instead of "root".

Also, do not hard-code the name "service".  Instead please pass it into this method as described in the comment below.